skrifa/src/outline/glyf/hint/engine/outline.rs

```diff
@@ -0,0 +1,218 @@
+    ///
+    /// Flips a range of points beginning with lowpoint and ending with highpoint so that
+    /// any off the curve points become on the curve points. The points are not marked as
+    /// touched.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#flip-range-on>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L5056>
+    pub(super) fn op_fliprgon(&mut self) -> OpResult {
+        self.do_fliprg(true)
+    }
+
+    /// Flip range off.
+    ///
+    /// FLIPRGOFF[] (0x82)
+    ///
+    /// Pops: highpoint: highest point number in range of points to be flipped (uint32)
+    ///       lowpoint: lowest point number in range of points to be flipped (uint32)
+    ///
+    /// Flips a range of points beginning with lowpoint and ending with
+    /// highpoint so that any on the curve points become off the curve points.
+    /// The points are not marked as touched.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#flip-range-off>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L5094>
+    pub(super) fn op_fliprgoff(&mut self) -> OpResult {
+        self.do_fliprg(false)
+    }
+
+    /// Untouch point.
+    ///
+    /// UTP[] (0x29)
+    ///
+    /// Pops: p: point number (uint32)
+    ///
+    /// Marks point p as untouched. A point may be touched in the x direction,
+    /// the y direction, both, or neither. This instruction uses the current
+    /// freedom_vector to determine whether to untouch the point in the
+    /// x-direction, the y direction, or both. Points that are marked as
+    /// untouched will be moved by an IUP (interpolate untouched points)
+    /// instruction. Using UTP you can ensure that a point will be affected
+    /// by IUP even if it was previously touched.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#untouch-point>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L6222>
+    pub(super) fn op_utp(&mut self) -> OpResult {
+        let p = self.value_stack.pop_usize()?;
+        let coord_axis = match (
+            self.graphics_state.freedom_vector.x != 0,
+            self.graphics_state.freedom_vector.y != 0,
+        ) {
+            (true, true) => Some(CoordAxis::Both),
+            (true, false) => Some(CoordAxis::X),
+            (false, true) => Some(CoordAxis::Y),
+            (false, false) => None,
+        };
+        if let Some(coord_axis) = coord_axis {
+            self.graphics_state.zp0_mut().untouch(p, coord_axis)?;
+        }
+        Ok(())
+    }
+
+    /// Helper for FLIPRGON and FLIPRGOFF.
+    fn do_fliprg(&mut self, on: bool) -> OpResult {
```

Comment:
`do_fliprg(true/false)` is pretty obtuse, an enum or a long name wouldn't hurt. Mildly surprisingly it doesn't seem to *flip* but rather to set so perhaps something like:

```rust
    /// Helper for FLIPRGON and FLIPRGOFF.
    fn set_on_curve_for_range(&mut self, on: bool) -> OpResult {
```